lms/djangoapps/courseware/courses.py

```diff
@@ -121,6 +122,16 @@ def check_course_access(course, user, action, check_if_enrolled=False):
 
     access_response = has_access(user, action, course, course.id)
     if not access_response:
+        # Redirect if StartDateError
+        if isinstance(access_response, StartDateError):
+            start_date = strftime_localized(course.start, 'SHORT_DATE')
+            params = QueryDict(mutable=True)
+            params['notlive'] = start_date
+            raise CourseAccessRedirect('{dashboard_url}?{params}'.format(
+                dashboard_url=reverse('dashboard'),
+                params=params.urlencode()
+            ))
```

Comment:
@andy-armstrong @dianakhuang: I decided to inline this instead of having the separate function call.  I think it reads better for now, and hopefully this will all be going away soon anyway as we streamline our redirection/messaging code.

I plan on merging as soon as tests are passing unless I hear otherwise.